components/shared/ResourceTag.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import workflowBadge from "@/public/icons/monitor.svg";
+import knowledgeBadge from "@/public/icons/greenbubble.svg";
+import componentBadge from "@/public/icons/numberlist.svg";
+import Image from "next/image";
+type ResourceTagType = "knowledge" | "component" | "workflow" | "plain";
+interface CustomBadgeProps extends React.PropsWithChildren {
+  type: ResourceTagType;
+  text?: string;
+}
+
+const ResourceTag = (props: CustomBadgeProps) => {
+  const { children, type, text } = props;
+  const badgeType = type || "plain";
+  const baseClass =
+    "inline-flex px-0.5 py-1 gap-1 rounded-[3px] items-center justify-center rounded-[3px]";
```

Comment:
```suggestion
    "inline-flex px-0.5 py-0.5 gap-1 rounded-[3px] items-center justify-center rounded-[3px]";
```
The y padding should be 2px